distribution/sdk/sample-mpegts-stream-generator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,277 @@
+                Duration videoDuration = getVideoDuration(videoFilePath);
+                tsDurationMillis = videoDuration.toMillis();
+            } catch (IOException | InterruptedException e) {
+                LOGGER.error("Unable to extract video duration from video : ", e);
+                return;
+            }
+            LOGGER.info("Video Duration : {}", tsDurationMillis);
+
+            double delayPerPacket = (double) tsDurationMillis / (double) tsPacketCount;
+
+            long startTime = System.currentTimeMillis();
+
+            int packetsSent = 0;
+
+            try (final FileInputStream fis = new FileInputStream(videoFile)) {
+
+                byte[] buffer = new byte[PACKET_SIZE];
+                int c;
+                while ((c = fis.read(buffer)) != -1) {
+                    bytesSent += c;
+
+                    ChannelFuture cf = ch.writeAndFlush(new DatagramPacket(Unpooled.copiedBuffer(
+                            buffer), new InetSocketAddress(ip, port)));
+
+                    cf.await();
+
+                    packetsSent++;
+
+                    if (packetsSent % 100 == 0) {
+                        Thread.sleep((long) (delayPerPacket * 100));
+                    }
+                    if (packetsSent % 10000 == 0) {
+                        LOGGER.info("packet sent={}", packetsSent);
+                    }
+                }
+            }
+
+            long endTime = System.currentTimeMillis();
+
+            LOGGER.info("elapsed: {}", endTime - startTime);
+
+            if (!ch.closeFuture()
+                    .await(100)) {
+                LOGGER.error("timeout");
+            }
+
+            LOGGER.info("bytes sent = " + bytesSent);
+
+        } catch (InterruptedException | IOException e) {
+            LOGGER.error("Unable to generate stream : ", e);
+        } finally {
+            // Shut down the event loop to terminate all threads.
+            eventLoopGroup.shutdownGracefully();
+        }
+
+        LOGGER.info("count = " + count.get());
+    }
+
+    private static CommandLine getFFmpegInfoCommand(final String videoFilePath) {
+        final String bundledFFmpegBinaryPath = getBundledFFmpegBinaryPath();
+        File file = new File("target/ffmpeg/" + bundledFFmpegBinaryPath);
+        return new CommandLine(file.getAbsolutePath()).addArgument(SUPPRESS_PRINTING_BANNER_FLAG)
+                .addArgument(INPUT_FILE_FLAG)
+                .addArgument(videoFilePath, DONT_HANDLE_QUOTING);
+    }
+
+    private static String getBundledFFmpegBinaryPath() {
+        if (SystemUtils.IS_OS_LINUX) {
+            return "linux/ffmpeg";
+        } else if (SystemUtils.IS_OS_MAC) {
+            return "osx/ffmpeg";
+        } else if (SystemUtils.IS_OS_SOLARIS) {
+            return "solaris/ffmpeg";
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            return "windows/ffmpeg.exe";
+        } else {
+            throw new RuntimeException("OS is not Linux, Mac, Solaris, or Windows."
```

Comment:
This isn't really input validation;  it's used for the ffmpeg command called from main.  I've cleaned up the error messages.